internal/controller/pkg/resolver/reconciler.go

```diff
@@ -268,22 +270,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{Requeue: false}, nil
 	}
 
+	n, _ := ndag.GetNode(dep.Identifier()) // nolint: errcheck // we know the node exists since it was implied
+
 	ref, err := name.ParseReference(dep.Package, name.WithDefaultRegistry(r.registry))
 	if err != nil {
 		log.Debug(errInvalidDependency, "error", err)
 		return reconcile.Result{Requeue: false}, nil
 	}
 
-	var addVer string
-	if addVer, err = r.findDependencyVersion(ctx, dep, log, ref); err != nil {
-		return reconcile.Result{Requeue: false}, errors.Wrap(err, errFindDependency)
-	}
-
 	// NOTE(hasheddan): consider creating event on package revision
 	// dictating constraints.
-	if addVer == "" {
+	addVer, err := r.versionFinder.FindValidDependencyVersion(ctx, dep, ref, n, log)
+	if err != nil || addVer == "" {
```

Comment:
Should we handle these separately? I assume there are different reasons to get a non-nil error vs getting `addVer` empty (supported with some comments on what that means). For example, the error we return doesn't include err, which could be possible if we handle separately.